host.go

```diff
@@ -0,0 +1,188 @@
+package f3
+
+import (
+	"bytes"
+	"context"
+	"time"
+
+	"github.com/filecoin-project/go-f3/gpbft"
+	"github.com/filecoin-project/go-f3/sim"
+	"golang.org/x/xerrors"
+)
+
+type gpbfthost struct {
+	participant *gpbft.Participant
+	manifest    Manifest
+	gpbft.Signer
+	gpbft.Verifier
+	broadcast func(context.Context, []byte) error
+
+	MessageQueue     chan *gpbft.GMessage
+	SelfMessageQueue chan *gpbft.GMessage //for the future when self messages are async
+
+	log Logger
+
+	timer      *time.Timer
+	runningCtx context.Context
+}
+
+func newHost(id gpbft.ActorID, m Manifest, broadcast func(context.Context, []byte) error,
+	s gpbft.Signer, v gpbft.Verifier, log Logger) (*gpbfthost, error) {
+	h := &gpbfthost{
+		manifest:         m,
+		Signer:           s,
+		Verifier:         v,
+		MessageQueue:     make(chan *gpbft.GMessage, 20),
+		SelfMessageQueue: make(chan *gpbft.GMessage, 20),
```

Comment:
Why the queue size?

Reply:
IDK seemed reasonable for the case when messages are coming in and we are running participants right now; I can make it a const.

Reply:
Just make sure there's no deadlock.

Reply:
There is no loopback from host to module/pubsub apart from Broadcast which doesn't go through the same loop.